### PR TITLE
Final PR to migrate region code to GRCh38

### PR DESCRIPTION
**Pull request by ch-kr**
(hopefully) last PR migrating region search code to GRCh38.

Major changes:
- Update docker image and remove option for custom machine in `rmc/pipeline/two_breaks/run_batches.py`

Minor changes:
- Remove `gnomad_qc` from requirements (looks like it's been added to gnomad-constraint requirements)
- Update number of desired partitions for constraint tables in `rmc/pipeline/regional_constraint.py` (also clean up some booleans and remove an obsolete TODO)
- Remove `parallel_file_exists` usage (function has been removed from gnomad_methods; `rmc/utils/simultaneous_breaks.py`)
- Remove dated cluster guidance in `rmc/pipeline/two_breaks/run_batches_dataproc.py`

### CODE REVIEW

File: rmc/pipeline/regional_constraint.py [hunk 1]
```diff
@@ -107,8 +114,7 @@ def main(args):
                 " significant break..."
             )
             if args.search_num == 1:
-                # TODO: Move existing HT to a different path so this can be remade
-                all_loci_chisq_ht_path = f"{SINGLE_BREAK_TEMP_PATH}/all_loci_chisq.ht"
+                all_loci_chisq_ht_path = f"{MODEL_PREFIX}/{CURRENT_GNOMAD_VERSION}/{args.freeze}/constraint_prep.ht"
```
Comment (lw453) [inline rmc/pipeline/regional_constraint.py:117]: To fix in accordance with PR 306

File: rmc/pipeline/two_breaks/run_batches.py [hunk 2]
```diff
@@ -609,23 +601,13 @@ def main(args):
...
-            args.docker_image = (
-                "us-central1-docker.pkg.dev/broad-mpg-gnomad/images/rmc_simul_search"
-            )
-            # NOTE: Python version on your local machine must match the Python version in this image
-            logger.warning(
-                "Using %s image; please make sure Hail and Python versions in image are"
-                " up to date",
-                args.docker_image,
-            )
+        args.docker_image = "hailgenetics/hail:0.2.132-py3.11"
```
Comment (lw453) [inline rmc/pipeline/two_breaks/run_batches.py:604]: The recommended image has `py3.10` in line 844

File: rmc/pipeline/two_breaks/run_batches.py [hunk 3]
```diff
@@ -839,15 +821,6 @@ def main(args):
             """,
         default="broad-mpg-gnomad",
     )
-    parser.add_argument(
-        "--use-custom-machine",
```
Comment (lw453) [inline rmc/pipeline/two_breaks/run_batches.py:843]: Line 682 still has reference to `use_custom_machine`

File: rmc/utils/simultaneous_breaks.py [hunk 4]
```diff
@@ -167,11 +166,9 @@ def get_sections_to_run(
     .. note::
         This step needs to be run locally due to permissions involved with `parallel_file_exists`.
```
Comment (lw453) [inline rmc/utils/simultaneous_breaks.py:167]: Now no longer in use

**Review by lw453 — CHANGES_REQUESTED** [review]

File: rmc/pipeline/regional_constraint.py (hunk 1 above)
Comment (ch-kr) [reply rmc/pipeline/regional_constraint.py:117]: going to leave this open in the hopes that we'll merge 306 first (and then I can merge those changes into this PR)

File: rmc/pipeline/two_breaks/run_batches.py (hunk 2 above)
Comment (ch-kr) [reply rmc/pipeline/two_breaks/run_batches.py:604]: oops!

File: rmc/utils/simultaneous_breaks.py (hunk 4 above)
Comment (ch-kr) [reply rmc/utils/simultaneous_breaks.py:167]: oh you're right -- we don't even use this function anymore. will delete

File: rmc/pipeline/regional_constraint.py (hunk 1 above)
Comment (ch-kr) [reply rmc/pipeline/regional_constraint.py]: done!

**Review by lw453 — APPROVED** [review]

**Pull request merged**